### ISSUE TITLE
Can't donwload the currentpic.cgi with curl

### ISSUE THREAD

**Issue by naitsimp** [EliasKotlyar/Xiaomi-Dafang-Hacks#803]
I try to download the currentpic.cgi from the camera with a curl command to a Raspberry Pi running home assistant (hassio). This used to worked but stopped suddenly.
I am executing the command on the pi directly via ssh.

My command:
`curl --insecure https://root:ismart12@192.168.1.249/cgi-bin/currentpic.cgi --output /config/www/dafang_1/dafang_1_single_snapshot.jpg -v`

Output:
```
core-ssh:~# curl --insecure https://root:ismart12@192.168.1.249/cgi-bin/currentpic.cgi --output /config/www/dafang_1/dafang_1_single_snapshot.jpg -v
*   Trying 192.168.1.249...
* TCP_NODELAY set
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:02:12 --:--:--     0* 
connect to 192.168.1.249 port 443 failed: Operation timed out
* Failed to connect to 192.168.1.249 port 443: Operation timed out
* Closing connection 0
curl: (7) Failed to connect to 192.168.1.249 port 443: Operation timed out
```

If I replace the URL to the camera with any picture URL from the web I am able to download the picture.
Running this command on the dafang itself results in a saved picture.
```
curl --insecure https://root:ismart12@192.168.1.249/cgi-bin
/currentpic.cgi --output /system/sdcard/DCIM/dafang_1_snapshot_door_closed_1.jpg -v
```

Running the command in cmd under Windows does also work, so I guess it is an issue with hassio.

Any idea what the problem might be? Thanks

**Comment by TungstenE2** [comment]
I have a differnt problem, but I assume our problems are related.

I use FHEM for my house automation.
If I try to get image by: https://root:ismart12@IP/cgi-bin/currentpic.cgi
I do not get a jpg file but a small file with extension svg. 

Same URL does work fine in browser. Image is displayed and has jpg format.

If I try a different webcam with same method a jpg snapshot is stored successfully by FHEM on my Raspberry Pi.

I have the feeling that something is wrong with the jpg format delivered by the dafang hack version, but I can not say what exactly. This is causing the problems we both have.

**Comment by funkypenguin** [comment]
I've noticed the same thing. Firmware from 2018-12-22T13:53:10Z works, while firmware 2018-12-30T04:56:09Z (and several inbetween) do not.

From my pi, I can successfully curl my 2018-12-22 dafang, but not my 2018-12-30 dafang (was previously on 2018-12-27 with the same issue)

From my macbook, I can curl either dafang without issue.

I can't even connect to port 80 with curl on the pi to the newer dafangs, to get the 401 HTTP header.

**Comment by jmtatsch** [comment]
You must use the https port.

**Comment by alex-phillips** [comment]
I'm getting a similar issue. The camera will be working fine, rtsp, https, etc. But everyone once in a while a device (my server) will no longer be able to connect. A simple curl to the https homepage times out while on my laptop I can access it just fine.

**Comment by alex-phillips** [comment]
I'm getting a similar issue. The camera will be working fine, rtsp, https, etc. But everyone once in a while a device (my server) will no longer be able to connect. A simple curl to the https homepage times out while on my laptop I can access it just fine.

**Comment by alex-phillips** [comment]
FWIW, during the times when I can't curl, I also can't SSH in from the same machine. But from a different machine on the SAME network, I am able to curl and SSH. I assume this is a device issue because this only happens sometimes. Restarting the camera makes both machines able to curl / ssh.

**Comment by liads** [comment]
I'm also experiencing this from time to time.
Not sure if it's related to the camera or to hassio. I've had cases in which one docker container couldn't connect to any port on the camera, but another container under the same hassio host connected without an issue.
Whenever I'm having this I need to restart both the camera and the hassio host to regain functionality.
It also happened consistently when I accidentally configured an incorrect password for the camera in HA.
My guess is it's some blacklisting issue, but I haven't yet been able to find where.

**Comment by TungstenE2** [comment]
I solved this by selling my Dafang. I really like the idea of the hack, but it is just not working reliable, and I have not seen active working on issues in the past months.

**Comment by alex-phillips** [comment]
@liads yeah if I could figure out if it was a process preventing these connections on the camera, I could run a cron to run an ssh command to restart whatever the problematic process is. But I can't seem to figure it out.

**Comment by smcallis77** [comment]
> @liads yeah if I could figure out if it was a process preventing these connections on the camera, I could run a cron to run an ssh command to restart whatever the problematic process is. But I can't seem to figure it out.

I have just upgraded one of my cameras to the rootfs version to try and eliminate the 'timeout' issue that has been raised in many issues including yours. I have four of these cameras running the latest version of the modified firmware, all Xiaomi Dafang 128MB models. All four have the issue from time to time - cannot ssh, cannot curl, cannot access the webserver (its intermittent and inconsistent). There have been various comments made around MQTT control/status being the cause but for me at least this is not the case. Certainly killing off any running processes i.e. MQTT status/control reduces the problem. But, I have killed all of the running processes I can leaving just 'dropbear' and I still get the issue albeit much less often. I also get a significant number of dropped packets which leads me to believe the issue is either in the wireless driver or the wireless connection itself or somewhere in the kernel. I have upgraded one of the cameras to rootfs and am still getting the dropouts and dropped packets - dang. Investigations continue. at least the rootfs version will allow me a better toolset to try to identify the issue.
